SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleaner.java

```diff
@@ -582,6 +616,29 @@ private void drainTaskCleanupQueue() {
     LOG.info("Killed {} tasks in {}", killedTasks, JavaUtils.duration(start));
   }
 
+  private void updateRequestToTaskMap(SingularityTaskCleanup cleanupTask, Map<String, List<String>> requestIdToTaskIds) {
+    SingularityTaskId taskId = cleanupTask.getTaskId();
+    List<String> requestTasks = requestIdToTaskIds.get(taskId.getRequestId());
+    if (requestTasks != null) {
+      requestTasks.add(taskId.getId());
+    } else {
+      requestTasks = new ArrayList<>(Collections.singletonList(taskId.getId()));
+      requestIdToTaskIds.put(taskId.getRequestId(), requestTasks);
+    }
+  }
+
+  private void cleanupRequestIfNoRemainingTasks(SingularityTaskId taskId, Map<String, List<String>> requestIdToTaskIds) {
+    List<String> requestTasks = requestIdToTaskIds.get(taskId.getRequestId());
+    requestTasks.remove(taskId.getId());
```

Comment:
`requestIdToTaskIds` isn't guaranteed to have the key present, will want to check that or this line will cause NPEs